src/core/types/index.js

```diff
@@ -23,3 +23,13 @@ export function isElement(value) {
 export function isFiniteNumber(value) {
   return typeof value === 'number' && isFinite(value);
 }
+
+/**
+ * Returns true if obj is a PromiseLike.
+ *
+ * @param {*} obj
+ * @return {obj is Promise<*>}
```

Comment:
```suggestion
 * @return {obj is PromiseLike<*>}
```

Reply:
Valid, but unless I'm missing something I'd consider this nonblocking as idk the practical difference between Promise and PromiseLike

Reply:
agreed, nonblocking